include/matx/core/tensor_utils.h

```diff
@@ -440,6 +440,40 @@ namespace detail {
     }
   }
 
+  /**
+   * convert Type to string
+   *
+   * function convert a tensor type to a string
+   *
+   */
+  template <typename T> static std::string GetTensorType()
+  {
+    if constexpr (std::is_same_v<T, int32_t>)
+      return "i4";
+    if constexpr (std::is_same_v<T, uint32_t>)
+      return "u4";
+    if constexpr (std::is_same_v<T, int64_t>)
+      return "i8";
+    if constexpr (std::is_same_v<T, uint64_t>)
+      return "u8";
+    if constexpr (std::is_same_v<T, float> || is_matx_half_v<T>)
+      return "f4";
+    if constexpr (std::is_same_v<T, double>)
+      return "f8";
+    if constexpr (std::is_same_v<T, cuda::std::complex<double>> ||
+                  std::is_same_v<T, std::complex<double>>) {
+      return "c8";
+    }
+    if constexpr (std::is_same_v<T, cuda::std::complex<float>> ||
+                  std::is_same_v<T, std::complex<float>> ||
+                  is_complex_half_v<T>) {
+      return "c4";
+    }
+
+    return "";
```

Comment:
Maybe do "Unknown" here?

Reply:
updated for all C++ types supported by MatX and added unknown